### PR TITLE
feat(appsync): queryDepthLimit and resolverCountLimit props on GraphqlApi

### PR DESCRIPTION
**Pull request by aerrober**
### Issue # (if applicable)

No open issue. Cr was based on https://github.com/aws/aws-cdk/pull/28500

### Reason for this change

Appsync added two additional fields documented in their CFN [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-resolvercountlimit) that were not previously supported in the L1 construct.

### Description of changes

Addition of field `queryDepthLimit` and `resolverCountLimit`

### Description of how you validated changes

Unit tests and integration tests were run locally. Additionally a sample app was deployed to validate these features.

### Checklist
- [ x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 1]
```diff
@@ -755,4 +755,28 @@ const api = new appsync.GraphqlApi(this, 'api', {
   definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.schema.graphql')),
   introspectionConfig: appsync.IntrospectionConfig.DISABLED,
 });
+```
+
+## Query Depth Limits
+
+There are times during which you may want more granular control over how the API functions during an operation. One such control is adding a limit to the amount of nested levels a query may process. By default, queries are able to process an unlimited amount of nested levels. Limiting queries to a specified amount of nested levels has potential implications for the performance and flexibility of your project.
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-appsync/README.md:762]: please split this up into a sentence per line so the raw markdown is more amenable to look at.

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 2]
```diff
@@ -755,4 +755,28 @@ const api = new appsync.GraphqlApi(this, 'api', {
...
+
+```ts
+const api = new appsync.GraphqlApi(this, 'api', {
+  name: 'LimitQueryDepths',
+  definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.schema.graphql')),
+  queryDepthLimit: 2
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-appsync/README.md:768]:
```suggestion
  queryDepthLimit: 2,
```

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 3]
```diff
@@ -755,4 +755,28 @@ const api = new appsync.GraphqlApi(this, 'api', {
...
+
+## Resolver Count Limits
+
+You can control how many resolvers each query can process. By default, each query can process up to 10000 resolvers. By setting a limit. AppSync will not handle any resolvers past a certain number limit.
+
+```ts
+const api = new appsync.GraphqlApi(this, 'api', {
+  name: 'LimitResolverCount',
+  definition: appsync.Definition.fromFile(path.join(__dirname, 'appsync.schema.graphql')),
+  resolverCountLimit: 2
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-appsync/README.md:780]:
```suggestion
  resolverCountLimit: 2,
```

File: packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts [hunk 4]
```diff
@@ -442,6 +442,20 @@ export interface GraphqlApiProps {
...
+   * A number indicating the maximum depth resolvers should be accepted when handling queries.
+   *
+   * @default - No Limit
+   */
+  readonly queryDepthLimit?: number;
+
+  /**
+   * A number indicating the maximum number of resolvers that should be accepted when handling queries.
+   *
+   * @default - No Limit
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts:456]: https://docs.aws.amazon.com/appsync/latest/APIReference/API_GraphqlApi.html

"The default value is 0 (or unspecified), which will set the limit to 10000"

File: packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts [hunk 5]
```diff
@@ -442,6 +442,20 @@ export interface GraphqlApiProps {
    * @default IntrospectionConfig.ENABLED
    */
   readonly introspectionConfig?: IntrospectionConfig;
+
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts:445]: both of these new props have lower and upper bounds according to the docs: https://docs.aws.amazon.com/appsync/latest/APIReference/API_GraphqlApi.html

They should be documented in their docstrings, checked for at synth time, and also have a unit test for each testing the boundary

**Review by kaizencc — DISMISSED** [review]
Thanks for the contribution @aerrober. Just some additions needed

File: packages/aws-cdk-lib/aws-appsync/README.md (hunk 1 above)
Comment (aerrober) [reply packages/aws-cdk-lib/aws-appsync/README.md]: Fixed

File: packages/aws-cdk-lib/aws-appsync/README.md (hunk 2 above)
Comment (aerrober) [reply packages/aws-cdk-lib/aws-appsync/README.md]: Fixed

File: packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts (hunk 4 above)
Comment (aerrober) [reply packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts]: Added

File: packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts (hunk 5 above)
Comment (aerrober) [reply packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts:445]: Added

**Review by kaizencc — APPROVED** [review]

**Review by kaizencc — APPROVED** [review]

**Pull request merged**